### ISSUE TITLE
Extension host terminated unexpectedly.

### ISSUE THREAD

**Issue by RustyWraith** [kortina/vscode-markdown-notes#135]
Enabling the extension causes the "Extension host terminated unexpectedly" error.
![image](https://user-images.githubusercontent.com/11280514/112157565-63f3bb80-8bf8-11eb-8d0a-f9cc94266863.png)

**Versions**
 Version: 1.55.0-insider
Commit: a6f6b4aa8e3a688d9a5cbcd5df14af511ce2bd8e
Date: 2021-03-23T05:28:25.637Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT ia32 10.0.18363

**Comment by kortina** [comment]
Anyone else having this issue?

**Comment by RustyWraith** [comment]
<details>
  <summary>Click to expand!</summary>
![image](https://user-images.githubusercontent.com/11280514/112167767-c7361b80-8c01-11eb-961e-03917d3ac20c.png)
![image](https://user-images.githubusercontent.com/11280514/112167850-d5843780-8c01-11eb-9332-f25b612c251a.png)
[log.log](https://github.com/kortina/vscode-markdown-notes/files/6190591/log.log)

</details>

**Comment by kortina** [comment]
From a quick look, it appears maybe you do not have `git` installed and this is a dependency? Can you confirm?

1. quickfix: install git
2. mid term: we can try to have a fallback, but this may really impact performance, so you prolly want to do (1) anyway

Surprised this has never come up before if this is the root cause...

**Comment by RustyWraith** [comment]
I've installed git system-wide and added path to git_path:
![image](https://user-images.githubusercontent.com/11280514/112183281-264e5d00-8c0f-11eb-9a07-737dcd8fcd52.png)
The error changed:

> findNonIgnoredFiles-git-exec-error Error: Command failed: git check-ignore
> fatal: not a git repository (or any of the parent directories): .git

The Extension crushes only if specific folder is being opened:
> D:\backup\Gdrive\LOGBOOK\Обработка_результатов
Could it be almighty Russian "I'll slip into you path and broke everything" Cyrillic?

**Comment by kortina** [comment]
Maybe. Do you still get a crash if you checkout the source for this repo and open the test dir: https://github.com/kortina/vscode-markdown-notes/tree/master/test

**Comment by RustyWraith** [comment]
Works fine.
![image](https://user-images.githubusercontent.com/11280514/112189691-4f71ec00-8c15-11eb-917d-26f212003bc7.png)
Some folders cause crashes:
![image](https://user-images.githubusercontent.com/11280514/112190811-66650e00-8c16-11eb-90b2-8beb276fd010.png)
I have no idea, what crashing folders have in common.

**Comment by kortina** [comment]
Maybe find 1 crashing folder, and then go through and remove all files except 1 for each file and see if you can isolate the crash to a specific file causing it?

**Comment by RustyWraith** [comment]
It's somehow connected with total size of .txt files.
If any (I didn't test all of them, only tested a few) .txt of size 20+MB is deleted, the error is gone.
![image](https://user-images.githubusercontent.com/11280514/112202277-3de31100-8c22-11eb-94da-f12fcc276049.png)

**Comment by krry** [comment]
I'm experiencing similar behavior, in that Markdown Notes seems to be crashing the extension host, but a different error message in the VSCode console.

<img width="831" alt="Screen Shot 2021-06-12 at 23 07 18" src="https://user-images.githubusercontent.com/439967/121797100-f17cf100-cbd2-11eb-86f7-df2ed91d301e.png">

I don't understand why it fails to find `/bin/sh` which is assuredly present.

When I disable Markdown Notes and reload the VSCode window, the extension host no longer crashes.

I'm installing Insiders now to test there as well.

**Comment by jeffmcneill** [comment]
Similar issue: 
![Screenshot from 2022-09-21 17-42-01](https://user-images.githubusercontent.com/3119827/191486343-3c2a2cf8-8561-4908-9f85-db3804f33c41.png)

crashing the extensions and VS code itself.

**Comment by kortina** [comment]
I'm unable to replicate this issue. 

@jeffmcneill what OS are you on? How many files and folders are in the project folder you're trying to use the extension with? (Maybe a project size issue similar to @RustyWraith's?)

**Comment by jeffmcneill** [comment]
@kortina 
- Debian Testing
- VS Code 1.71.2
- Project folder has a large number of files and folders, as I use this as a general text editor. Hundreds of folders, thousands of files. Does the extension try and read those files/folders in some way?

**Comment by kortina** [comment]
Yes, it parses the contents of every file to find backlinks and tags. It probably won't work for such a large project folder. Long term it probably makes more sense to move all of this to a language server / background process, but that's not the way it works today. Sorry!
